### PR TITLE
Db/fix core memcheck

### PR DESCRIPTION
**Pull request by derekwbrown**
### What does this PR do?

fixes core memory check to report all stats that A5 reported.

### Motivation

desire to have A6 match A5

### Additional Notes

Anything else we should know when reviewing?

### CODE REVIEW

**Review by remicalixte — DISMISSED** [review]

**Comment by remicalixte** [comment]
All the metrics are working great!

File: pkg/collector/corechecks/system/cpu_windows.go [hunk 1]
```diff
@@ -98,8 +100,14 @@ func (c *CPUCheck) Run() error {
 		sender.Gauge("system.cpu.idle", idle*toPercent, "", nil)
```
Comment (remicalixte) [inline pkg/collector/corechecks/system/cpu_windows.go]: Would we want to use pdh here too, to get these metrics ?

File: pkg/collector/corechecks/system/memory_windows.go [hunk 2]
```diff
@@ -25,17 +28,72 @@ var runtimeOS = runtime.GOOS
...
+			log.Warnf("Could not retrieve value for system.mem.paged")
+		}
+	}
+
+	if c.nonpagedBytes != nil {
+		val, err = c.nonpagedBytes.GetSingleValue()
+		if err == nil {
+			sender.Gauge("system.mem.nonpaged", float64(val)/mbSize, "", nil)
+		} else {
+			log.Warnf("Could not retrieve value for system.mem.nonpaged")
```
Comment (olivielpeau) [inline pkg/collector/corechecks/system/memory_windows.go]: Intuitively I'd say it'd be useful to add the error to the logged message, in order to have more info on what failed exactly

(applies to all the logging calls added in this function)

Comment (olivielpeau) [reply pkg/collector/corechecks/system/memory_windows.go]: also, the logic at the end of the function:

```
if errVirt != nil && errSwap != nil {
    return fmt.Errorf("failed to gather any memory information")
}
```

needs to be updated

File: releasenotes/notes/fixcorememcheck-7249771b6fdd38fe.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    Adds system.cpu.interrupt, and system.mem.committed, system.mem.paged,
+    system.mem.nonpaged, system.mem.cached metrics.
```
Comment (olivielpeau) [inline releasenotes/notes/fixcorememcheck-7249771b6fdd38fe.yaml]: we should mention `on Windows` here

**Review by olivielpeau — COMMENTED** [review]
Added a few nits, nothing major though

File: pkg/collector/corechecks/system/memory_windows.go [hunk 4]
```diff
@@ -25,17 +28,76 @@ var runtimeOS = runtime.GOOS
...
 	}
 
+	var val float64
+	if c.cacheBytes != nil {
+		val, err = c.cacheBytes.GetSingleValue()
+		if err == nil {
+			sender.Gauge("system.mem.cached", float64(val)/mbSize, "", nil)
+		} else {
+			log.Warnf("Could not retrieve value for system.mem.cached %v", err)
+			return err
```
Comment (olivielpeau) [inline pkg/collector/corechecks/system/memory_windows.go]: I think we'd want to attempt to collect the other metrics even if for some reason we encountered an error on this particular metric. Maybe simply removing the `return` statement is enough...

(applies to the other metrics as well)

**Review by olivielpeau — COMMENTED** [review]
one more comment, just to make sure a failure on one metric doesn't break the whole check

File: pkg/collector/corechecks/system/cpu_windows.go (hunk 1 above)
Comment (olivielpeau) [reply pkg/collector/corechecks/system/cpu_windows.go]: (feel free to create a card in our backlog :) )

**Review by olivielpeau — APPROVED** [review]
LGTM 👌 Feel free to merge once CI is 💚

**Pull request merged**